server/implementation/src/main/java/io/smallrye/graphql/execution/event/EventEmitter.java

```diff
@@ -19,22 +23,31 @@
  * @author Phillip Kruger (phillip.kruger@redhat.com)
  */
 public class EventEmitter {
-
-    private final ServiceLoader<EventingService> eventingServices = ServiceLoader.load(EventingService.class);
+    private static final Logger LOG = Logger.getLogger(EventEmitter.class);
+    private static final Map<Config, EventEmitter> INSTANCES = new IdentityHashMap<>();
     private final List<EventingService> enabledServices = new ArrayList<>();
 
-    public EventEmitter(Config config) {
+    public static EventEmitter getInstance(Config config) {
+        return INSTANCES.computeIfAbsent(config, k -> new EventEmitter(config));
+    }
+
+    private EventEmitter(Config config) {
+        ServiceLoader<EventingService> eventingServices = ServiceLoader.load(EventingService.class);
         Iterator<EventingService> it = eventingServices.iterator();
         while (it.hasNext()) {
+            EventingService eventingService = null;
             try {
-                EventingService eventingService = it.next();
+                eventingService = it.next();
                 String configKey = eventingService.getConfigKey();
                 boolean enabled = config.getConfigValue(configKey, boolean.class, false);
                 if (enabled) {
                     enabledServices.add(eventingService);
                 }
             } catch (Throwable t) {
                 // Ignore that service...
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Failed to register eventing service, " + eventingService, t);
```

Comment:
Shouldn't this rather be `WARN`? This looks like quite a serious problem (metrics and tracing won't work, for example)